akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -179,7 +175,21 @@ object ShardRegion {
   /**
    * Reply to `GetCurrentRegions`
    */
-  @SerialVersionUID(1L) final case class CurrentRegions(regions: Set[Address]) {
+  @SerialVersionUID(2L) final case class CurrentRegions(regionActorPaths: Set[ActorPath]) {
```

Comment:
My point is that `ActorPath` is too easy to use. They will send `GetCurrentShardRegionState` to the actor paths and do the aggregation at one central place (and thereby sending all ids over the net). I think this query should be as before, i.e. handing out addresses.

Then we could add another convenient aggregation query that return the full summary of everything, but only counts.